src/metabase/db/data_migrations.clj

```diff
@@ -172,6 +177,39 @@
                                  [:like
                                   :dashcard.visualization_settings "%\"click_link_template\":%"]]})))
 
+(defn- remove-group-id-from-mappings-by-setting-key
+   [mapping-setting-key admin-group-id]
+   ;; Intentionally get setting using `db/select-one-field` instead of `setting/get` because for some reasons
+   ;; during start-up setting/get return the default value defined in defsetting instead of value from Setting table
+   (let [mapping (try
+                  (json/parse-string (db/select-one-field :value Setting :key (name mapping-setting-key)))
+                  (catch Exception _e
+                    {}))]
+     (when-not (empty? mapping)
+       (setting/set-value-of-type!
+        :json mapping-setting-key
+        (into {}
+              (map (fn [[k v]] [k (filter #(not= admin-group-id %) v)]))
+              mapping)))))
+
+ (defmigration ^{:author "qnkhuat" :added "0.43.0"} migrate-remove-admin-from-group-mapping-if-needed
+   ;;  In the past we have a setting to disable group sync for admin group when using SSO or LDAP, but it's broken and haven't really worked (see #13820)
+   ;;  In #20991 we remove this option entirely and make sync for admin group just like a regular group.
+   ;;  But with this change we want to make sure we don't accidently add/remove adminusers we need to :
+   ;;  - for LDAP, if the `ldap-sync-admin-group` is disabled, remove all mapping for admin group
+   ;;  - for SAML, JWT remove all mapping for admin group
+   (let [admin-group-id (u/the-id (group/admin))]
+     (when (and (ldap/ldap-configured?)
+                (= "false" (db/select-one-field :value Setting :key "ldap-sync-admin-group")))
+       (remove-group-id-from-mappings-by-setting-key :ldap-group-mappings admin-group-id))
+     ;; sso are enterprise feature, so only remove admin groups if enterprise available
+     (when (premium-features/enable-sso?)
```

Comment:
eh, I'm starting to think we should just run this logic for OSS instances too. maybe this is a super small edge case but if someone was running EE and set up an admin group mapping, then downgraded to OSS and ran this migration, then upgraded back to EE, we probably don't want to suddenly start syncing the admin group unless they add it again

Reply:
I changed it so that we’ll run it even in oss, the check conditions -are similar to jwt-configured? And saml-configured?

I suppose if a customer change from ee to oss, those settings will not be overridden right? I.e: automatically change saml-enabled to false?